### PR TITLE
Fix chemicals not recognizing end of metabolization

### LINKED ISSUE #53546: Zombie/Ghoul Powder Fake Death Becomes Perma Death
## Reproduction:

1). Give some poor sod `/toxin/zombiepowder` or `/toxin/ghoulpowder`.
2). `Trait fake_death` and henceforth `trait death_coma` are applied.
3). Remove the reagent from their system through the miracles of modern medicine.
4). `Trait fake_death` and `trait death_coma` are retained.
5). They are now perma dead unless the traits are removed through admin means.

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Fixes #53546.

Also as an aside thing, adds `TEST_FOCUS` to make it easy during debugging to focus just on the test you're writing rather than seeing all the extra noise.

## Why It's Good For The Game
A lot of chemicals do things when they're out of your system. These effects should work.

## Changelog
:cl:
fix: Chemicals now correctly do effects when they stop metabolizing.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -614,12 +614,20 @@
...
 		if(R.type == reagent)
-			if(my_atom && isliving(my_atom))
-				var/mob/living/M = my_atom
+			var/mob/living/mob_consumer
+
+			if (isliving(my_atom))
+				mob_consumer = my_atom
+			else if (istype(my_atom, /obj/item/organ/stomach))
+				var/obj/item/organ/stomach/belly = my_atom
+				mob_consumer = belly.owner
```
Comment (Jared-Fogle) [inline code/modules/reagents/chemistry/holder.dm:623]: I hate this. Let me know if there's anything better.

Comment (nightred) [reply code/modules/reagents/chemistry/holder.dm:623]: holder should just proc/get_holder_mob() and return the mob for any organ or container like this
and the check could be for all organs not just stomach as i think of it.

**Comment by ATH1909** [comment]
called it

**Comment by Winterous** [comment]
This explains why my ephedrine movespeed buff didn't wear off.

File: code/modules/reagents/chemistry/holder.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/holder.dm:623]: any reason why this is specific to stomachs?

Comment (Jared-Fogle) [reply code/modules/reagents/chemistry/holder.dm:623]: I can make it check just for organs, if that typecheck still seems reasonable to you.

Comment (nightred) [reply code/modules/reagents/chemistry/holder.dm:623]: this is fine as is, i am just overthinking it and ensuring no organ has a reagent that could be acting on you

**Pull request merged**